pkg/bmc/virtualmedia_validation_test.go

```diff
@@ -0,0 +1,143 @@
+package bmc
+
+import (
+	"testing"
+
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/testserver"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	// NoManagedBy represents a System without ManagedBy
+	NoManageBy string = "NoManageBy"
+	// HasManageBy represents a System with ManagedBy
+	HasManageBy string = "ManageBy"
+
+	// NoVirtualMedia represents a Manager without VirtualMedia
+	NoVirtualMedia string = "NoVirtualMedia"
+	// HasVirtualMedia represents a Manager with VirtualMedia
+	HasVirtualMedia string = "VirtualMedia"
+)
```

Comment:
These constants are indicating some pre-packaged answers (payload), but they really should be prepared in builder functions (or even better types) defined outside the mock (note that the original idea of reusing JSON struct was meant just to address this specific problem).  The goal is to have a readable setup and promote reuse as much as possible - and for that having general Mock functions 1:1 with the API definition.

As an alternative (to the lack of JSON struct) I'd suggest to give a look at the [hostBuilder](https://github.com/metal3-io/baremetal-operator/blob/5ebf0ab6e367a854c4177edc6612381b1d65dc01/controllers/metal3.io/host_state_machine_test.go#L468-L567) example, since it adopts a fluent builder pattern to define an object used as input of the tests (ie, see [here](https://github.com/metal3-io/baremetal-operator/blob/5ebf0ab6e367a854c4177edc6612381b1d65dc01/controllers/metal3.io/host_state_machine_test.go#L48))

Reply:
Not sure if I understood everything, are you proposing we create some struct  (probably more than one, to represent each Redfish property - System/Manager/VirtualMedia )that can be used as parameter for the `RedfishMock` to return the correct JSON response?

Reply:
@andfasano hey, I did some changes and removed the constants, let me know if it makes sense ( I didn't move anything from the VirtualMedia in `pkg/provisioner/ironic/testserver/redfish.go` , maybe it would make sense to move to the test? or maybe move what I did in the test to testserver.)